### PR TITLE
FRONT/MOBILE: add login / register with redirection

### PR DESCRIPTION
**Pull request by badidu09**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [x] Do we need to implement unit tests?

* **In what state is your pull request?**
- [ ] Ready to merge / Waiting a reviwer to see my work.
- [x] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [x] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `Layout` component utilizing the `expo-router` `Stack` for navigation.
  - Added login and index screens with hidden header for improved user experience.
  - Implemented state management for login screen with form submission and error handling.

- **Enhancements**
  - Improved user interface with new styled components for a more visually appealing design.

- **Bug Fixes**
  - Added error handling mechanisms in the login process to provide feedback and guidance to users.

- **Documentation**
  - Updated comments and documentation to reflect the addition of new components and navigation flow.

### CODE REVIEW

**Review by romainpanno — DISMISSED** [review]
change code rbbit requests

File: mobile/app/login.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+  const router = useNavigation();
+
+  useEffect(() => {
+    if (loginSuccess) {
+      // Redirect to the home page
+      router.navigate("home" as never);
+    }
+  }
+  , [loginSuccess, router]);
```
[the remark this answers is not in the document]
Comment (badidu09) [reply mobile/app/login.tsx]: what I have to do ?

File: mobile/app/_layout.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Stack } from 'expo-router';
+import login from './login';
+import index from './index';
+
+
```
Comment (Vivelis) [inline mobile/app/_layout.tsx:5]: Should be only one empty line

```suggestion

```

File: mobile/app/login.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      // Any other clean-up actions can be placed here
+    };
+  }, [loginSuccess, router]);
+
+  const handleError = (error: any) => {
+    console.error("Error:", error);
+    setError("An error occurred, please try again");
+  };
+
+
```
Comment (Vivelis) [inline mobile/app/login.tsx:40]: Should be only one empty line.

```suggestion

```

**Review by Vivelis — CHANGES_REQUESTED** [review]

**Review by Vivelis — APPROVED** [review]

**Review by KitetsuK — APPROVED** [review]

**Pull request merged**